modules/programs/gitui/gitui.nix

```
@@ -0,0 +1,74 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+
+  cfg = config.programs.gitui;
+
+  yamlFormat = pkgs.formats.yaml { };
+
+  inherit (pkgs.stdenv.hostPlatform) isDarwin;
+
+in {
+  meta.maintainers = [ maintainers.mifom ];
+
+  options.programs.gitui = {
+    enable =
+      mkEnableOption "gitui, blazing fast terminal-ui for git written in rust";
+
+    key_config = mkOption {
+      type = types.str;
+      default = "";
+      example = ''
+        exit: ( code: Char('c'), modifiers: ( bits: 2,),),
+        quit: ( code: Char('q'), modifiers: ( bits: 0,),),
+        exit_popup: ( code: Esc, modifiers: ( bits: 0,),),
+      '';
+      description = ''
+        Key config in Ron file format. It written to
+        <filename>~/.config/gitui/key_config.ron</filename>.
+      '';
+    };
+    theme = mkOption {
+      type = types.str;
+      default = ''
+        (
+          selected_tab: Reset,
+          command_fg: White,
+          selection_bg: Blue,
+          cmdbar_extra_lines_bg: Blue,
+          disabled_fg: DarkGray,
+          diff_line_add: Green,
+          diff_line_delete: Red,
+          diff_file_added: LightGreen,
+          diff_file_removed: LightRed,
+          diff_file_moved: LightMagenta,
+          diff_file_modified: Yellow,
+          commit_hash: Magenta,
+          commit_time: LightCyan,
+          commit_author: Green,
+          danger_fg: Red,
+          push_gauge_bg: Blue,
+          push_gauge_fg: Reset,
+        )
+      '';
+      description = ''
+        Theme in Ron file format. It written to
+        <filename>~/.config/gitui/theme.ron</filename>.
+      '';
+    };
+  };
+
+  config = mkIf cfg.enable {
+    home.packages = [ pkgs.gitui ];
+
+    xdg.configFile."gitui/theme.ron" = { text = cfg.theme; };
```

```suggestion
    xdg.configFile."gitui/theme.ron".source = if (builtins.isPath cfg.theme || lib.isStorePath cfg.theme) then cfg.theme else pkgs.writeText "gitui-theme.ron" cfg.theme;
```

This allows using either a file or a string. The code will need to be formatted with the formatting script.